src/components/Post/Post.tsx

```diff
@@ -41,60 +59,71 @@ const Post: React.FC<PostProps> = ({ stateStore }) => {
     };
 
     const updateImage = () => {
-        setPostRect(image, stateStore.screenInfo.width, stateStore.screenInfo.height, stateStore);
+        setPostRect(image, screenInfo.width, screenInfo.height, visiblePost);
     };
 
     image.onload = () => {
         updateImage();
+        setImageHeight(imageRef.current?.clientHeight ?? 0);
     };
 
     useEffect(() => {
-        stateStore.clearVisiblePostTagNames();
-        getPost(parseInt(postId), stateStore.siteInfo.siteUrl)
-            .then(post => {
-                stateStore.setCurrentPost(post);
+        let tagNames: string[] = [];
+        let url = '';
+        clearVisiblePostTagNames();
+        getPost(parseInt(postId), siteInfo.siteUrl)
+            .then((post: PostState) => {
+                return post;
             })
-            .then(() => {
-                getTagNames(stateStore.visiblePost.tags, stateStore.siteInfo.siteUrl)
+            .then((newPost: PostState) => {
+                getTagNames(newPost.tags, siteInfo.siteUrl)
                     .then(tags => {
-                        stateStore.setVisiblePostTags(tags);
+                        tagNames = tags;
                     });
-                getPostImage(stateStore.visiblePost.featured_media, stateStore.siteInfo.siteUrl)
-                    .then(imageUrl => {
-                        stateStore.setVisiblePostImage(imageUrl);
+                getPostImage(newPost.featured_media, siteInfo.siteUrl)
+                    .then((imageUrl: string) => {
+                        url = imageUrl;
+                        setCurrentPost(newPost, tagNames, url);
                     });
             });
     }, [postId]);
 
     useEffect(() => {
-        if (stateStore.visiblePost.fullImageUrl !== null && stateStore.visiblePost.fullImageUrl !== undefined) {
-            image.src = stateStore.visiblePost.fullImageUrl;
+        if (visiblePost.fullImageUrl !== null && visiblePost.fullImageUrl !== undefined) {
```

Comment:
Couldn't you just use `!visiblePost.fullImageUrl` instead or do you expect that it might get an empty string. What happens in that case? Do you not want to include it in the condition for that as well?

Reply:
Yes, I think this can just be `if (visiblePost.fullImageUrl) {`.

If it does get an empty string, this is also `falsy` so it will not enter the `if`.

Reply:
Changed.